core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -695,12 +710,21 @@ object Stream {
 
   implicit def covaryPurePipe2[F[_],I,I2,O](p: Pipe2[Pure,I,I2,O]): Pipe2[F,I,I2,O] = pipe2.covary[F,I,I2,O](p)
 
-  implicit def streamCatchableInstance[F[_]]: Catchable[({ type λ[a] = Stream[F, a] })#λ] =
-    new Catchable[({ type λ[a] = Stream[F, a] })#λ] {
+  implicit def streamMonadErrorInstance[F[_]]: MonadError[Stream[F,?], Throwable] =
+    new MonadError[Stream[F,?], Throwable] {
```

Comment:
No, the compiler plugin is not necessary for users -- it is compile time only. Users are free to use type lambda or type aliases.